### PR TITLE
feat: Adding SSO support for Okta. Fixes #6165

### LINKED ISSUE #6165: Groups are not working with okta sso integration
## Summary 
We're running Argo Workflows with SSO and RBAC enabled. We're using Okta as the OIDC provider.

Our service-account, which pointed to RoleBinding:
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: argo-user
  annotations:
    workflows.argoproj.io/rbac-rule: "'admins' in groups"
    workflows.argoproj.io/rbac-rule-precedence: "1"
```
Our config-map: 
```
data:
  config: |
    sso:
      clientID:
        key: clientID
        name: argo-secret
      clientSecret:
        key: clientSecret
        name: argo-secret
      issuer: https://<out_okta_url>
      name: Okta
      rbac:
        enabled: true
      redirectUrl: https://<our_argo_url>/oauth2/callback
      scopes:
      - groups
      - email
```

When we try to set a rule for the email `email endsWith '@ourdomain.com'`, rbac works as expected and with groups in doesn't work.
And we have `okta.groups.read` granted in Okta app. 

Expected behaviour that groups rules to work for okta.

## Diagnostics
We are using EKS.
Argo Workflows v3.0.7.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by gageorsburn**
…d with Okta

Signed-off-by: Gage Orsburn <gorsburn@botkeeper.com>

Tips:

* Maybe add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).
* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.
* If changes were requested, and you've made them, then dismis the review to get it looked at again.
* You can ask for help!

### CODE REVIEW

**Comment by gageorsburn** [comment]
This should fix #6165. Lemme know if there are any changes or additional things I need to do.

File: USERS.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ Currently, the following organizations are **officially** using Argo Workflows:
 1. [Woolworths Group](https://www.woolworthsgroup.com.au/)
 1. [Workiva](https://www.workiva.com/)
 1. [Zhihu](https://www.zhihu.com/)
-
+1. [Botkeeper](https://www.botkeeper.com/)
```
Comment (alexec) [inline USERS.md:145]: alphabetical please!

File: server/auth/types/claims.go [hunk 2]
```diff
@@ -62,3 +77,28 @@ func (c *Claims) GetCustomGroup(customKeyName string) ([]string, error) {
 
 	return newSlice, nil
 }
+
+func (c *Claims) GetUserInfoGroups(accessToken, issuer, userInfoPath string) ([]string, error) {
+	url := fmt.Sprintf("%s%s", issuer, userInfoPath)
+	request, _ := http.NewRequest("GET", url, nil)
```
Comment (alexec) [inline server/auth/types/claims.go:83]: please return error

**Review by alexec — CHANGES_REQUESTED** [review]
two small changes

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
Just FYI - we don't test your changes, we don't use Okta. It'd be good to know how you tested this please.

**Comment by gageorsburn** [comment]
@alexec I have one of our Argo workflows environments set up with Okta. I built a custom image with only the changes in the pull request. After adding the configuration flag in just signing into Argo Workflows and verifying the proper groups exist.

**Pull request merged**